src/Elf.cpp

```diff
@@ -772,19 +773,54 @@ std::vector<char> write_shared_object_internal(Object &obj, Linker *linker, cons
     hash.set_alignment(4);
     hash.set_flag(Section::SHF_ALLOC);
     size_t sym_count = syms.size();
+    std::vector<unsigned long> sym_hashes;
```

Comment:
The logic I've seen used in LLVM's linker is to simply set bucket_count = sym_count. That would be a lot simpler :)